conftest.py

```diff
@@ -83,6 +87,44 @@ def tmpdir_function(request, tmpdir_class, monkeypatch):
         monkeypatch.setenv('TMPDIR', tmpdir)
         yield tmpdir
 
+@pytest.fixture
```

Comment:
I don't think this should be a fixture and contextmanager. The pytest philosophy doesn't want to use `with mock(obj) as m...` lines in the test cases because it breaks up the flow. https://pypi.org/project/pytest-mock/. For your actual test on the CPU counts the simple `mock.return_value = '1'` would suffice. You can perhaps monkeypatch the `MockFixture` module to add this functionality if necessary.

Reply:
The available_cpu_count tests needs to monkeypatch different return values of slurp_file for different arguments.  Also, a contextmanager is needed if you want to test more than one thing within one test function.  monkeypatch with context managers is built into pytest, so pytest philosophy does allow for that.